app/utils/callback/lava.py

```diff
@@ -640,6 +640,42 @@ def _add_rootfs_info(group, base_path, file_name="build_info.json"):
         utils.LOG.warn("ValueError: {}".format(e))
 
 
+def _adjust_log_line_numbers(results, log):
+    """ A workaround for a race condition effect visible in LAVA callbacks
+
+    This function changes log end line numbers in LAVA test results to match
+    the moment when the signal was sent instead of when it was received.
+    Function doesn't return value, but modifies results dictionary.
+    """
+    signal_rcvd_pattern = re.compile(r'Received signal: '
```

Comment:
This could be moved as a module attribute, to be compiled only once rather than every time the function is called.

Reply:
fixed

Reply:
turki511ghbnm